### ISSUE TITLE
Flow could not be retrieved from deployment with s3fs==2023.3.0

### ISSUE THREAD

**Issue by LeeMendelowitz** [PrefectHQ/prefect#8710]
### First check

- [X] I added a descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the Prefect documentation for this issue.
- [X] I checked that this issue is related to Prefect and not one of its dependencies.

### Bug summary

I started to get a "Flow could not be retrieved from deployment" error for a flow run deployment which had been previously working.

After several hours of investigation, I tracked it down a trailing slash required for `S3.bucket_path` with dependency `s3fs==2023.3.0`.  A trailing slash was not required under `s3fs==2023.1.0`. 

By simply adding a trailing slash to the `S3.bucket_path`, everything seems to work again.

### Reproduction

```python3
from prefect import flow, get_run_logger
from prefect.deployments import Deployment
from prefect.filesystems import S3
from prefect.infrastructure import Process
from pathlib import Path

@flow
def my_flow():
    logger = get_run_logger()
    logger.info("Hello")
    return True

# The deployment flow run fails without a trailing slash in the `bucket_path` on s3fs==2023.3.0
# The deployment flow run succeeds without a trailing slash in the `bucket_path` on s3fs==2023.1.0
storage = S3(bucket_path = 'my-bucket/test-flow')

working_dir = Path('./working_dir').resolve()
working_dir.mkdir(exist_ok = True)
infrastructure = Process(working_dir=working_dir)

my_deployment = Deployment.build_from_flow(
    flow = my_flow,
    name = 'my-deployment',
    output = 'deployment.yaml',
    storage = storage,
    infrastructure = infrastructure,
    skip_upload = True,
    apply = False
)

if __name__ == '__main__':
    my_deployment.apply(upload = True)
```

### Error

```python3
Downloading flow code from storage at None

Flow could not be retrieved from deployment.
Traceback (most recent call last):
  File "<frozen importlib._bootstrap_external>", line 879, in exec_module
  File "<frozen importlib._bootstrap_external>", line 1016, in get_code
  File "<frozen importlib._bootstrap_external>", line 1073, in get_data
FileNotFoundError: [Errno 2] No such file or directory: '/Users/lmendelowitz/playground/prefect/storage_issue/working_dir/deployment.py'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Users/lmendelowitz/.local/share/virtualenvs/storage_issue-SDGII8WY/lib/python3.10/site-packages/prefect/engine.py", line 270, in retrieve_flow_then_begin_flow_run
    flow = await load_flow_from_flow_run(flow_run, client=client)
  File "/Users/lmendelowitz/.local/share/virtualenvs/storage_issue-SDGII8WY/lib/python3.10/site-packages/prefect/client/utilities.py", line 47, in with_injected_client
    return await fn(*args, **kwargs)
  File "/Users/lmendelowitz/.local/share/virtualenvs/storage_issue-SDGII8WY/lib/python3.10/site-packages/prefect/deployments.py", line 187, in load_flow_from_flow_run
    flow = await run_sync_in_worker_thread(load_flow_from_entrypoint, str(import_path))
  File "/Users/lmendelowitz/.local/share/virtualenvs/storage_issue-SDGII8WY/lib/python3.10/site-packages/prefect/utilities/asyncutils.py", line 91, in run_sync_in_worker_thread
    return await anyio.to_thread.run_sync(
  File "/Users/lmendelowitz/.local/share/virtualenvs/storage_issue-SDGII8WY/lib/python3.10/site-packages/anyio/to_thread.py", line 31, in run_sync
    return await get_asynclib().run_sync_in_worker_thread(
  File "/Users/lmendelowitz/.local/share/virtualenvs/storage_issue-SDGII8WY/lib/python3.10/site-packages/anyio/_backends/_asyncio.py", line 937, in run_sync_in_worker_thread
    return await future
  File "/Users/lmendelowitz/.local/share/virtualenvs/storage_issue-SDGII8WY/lib/python3.10/site-packages/anyio/_backends/_asyncio.py", line 867, in run
    result = context.run(func, *args)
  File "/Users/lmendelowitz/.local/share/virtualenvs/storage_issue-SDGII8WY/lib/python3.10/site-packages/prefect/flows.py", line 809, in load_flow_from_entrypoint
    flow = import_object(entrypoint)
  File "/Users/lmendelowitz/.local/share/virtualenvs/storage_issue-SDGII8WY/lib/python3.10/site-packages/prefect/utilities/importtools.py", line 197, in import_object
    module = load_script_as_module(script_path)
  File "/Users/lmendelowitz/.local/share/virtualenvs/storage_issue-SDGII8WY/lib/python3.10/site-packages/prefect/utilities/importtools.py", line 160, in load_script_as_module
    raise ScriptError(user_exc=exc, path=path) from exc
prefect.exceptions.ScriptError: Script at 'deployment.py' encountered an exception: FileNotFoundError(2, 'No such file or directory')
```

### Versions

```Text
Version:             2.8.3
API version:         0.8.4
Python version:      3.10.8
Git commit:          07de1ac4
Built:               Thu, Feb 23, 2023 4:31 PM
OS/Arch:             darwin/arm64
Profile:             local
Server type:         server
```

### Additional context

Here are the requirements.txt:

```
-i https://pypi.org/simple
aiobotocore==2.4.2 ; python_version >= '3.7'
aiohttp==3.8.4 ; python_version >= '3.6'
aioitertools==0.11.0 ; python_version >= '3.6'
aiosignal==1.3.1 ; python_version >= '3.7'
aiosqlite==0.18.0 ; python_version >= '3.7'
alembic==1.10.0 ; python_version >= '3.7'
anyio==3.6.2 ; python_full_version >= '3.6.2'
appnope==0.1.3 ; sys_platform == 'darwin'
apprise==1.3.0 ; python_version >= '3.6'
asgi-lifespan==2.0.0 ; python_version >= '3.7'
asttokens==2.2.1
async-timeout==4.0.2 ; python_version >= '3.6'
asyncpg==0.27.0 ; python_full_version >= '3.7.0'
attrs==22.2.0 ; python_version >= '3.6'
backcall==0.2.0
botocore==1.27.59 ; python_version >= '3.7'
cachetools==5.3.0 ; python_version ~= '3.7'
certifi==2022.12.7 ; python_version >= '3.6'
cffi==1.15.1
charset-normalizer==3.0.1
click==8.1.3 ; python_version >= '3.7'
cloudpickle==2.2.1 ; python_version >= '3.6'
colorama==0.4.6 ; python_version >= '2.7' and python_version not in '3.0, 3.1, 3.2, 3.3, 3.4, 3.5, 3.6'
coolname==2.2.0
croniter==1.3.8 ; python_version >= '2.6' and python_version not in '3.0, 3.1, 3.2, 3.3'
cryptography==39.0.2 ; python_version >= '3.6'
dateparser==1.1.7 ; python_version >= '3.7'
decorator==5.1.1 ; python_version >= '3.5'
docker==6.0.1 ; python_version >= '3.7'
executing==1.2.0
fastapi==0.92.0 ; python_version >= '3.7'
frozenlist==1.3.3 ; python_version >= '3.7'
fsspec==2023.3.0 ; python_version >= '3.8'
google-auth==2.16.2 ; python_version >= '2.7' and python_version not in '3.0, 3.1, 3.2, 3.3, 3.4, 3.5'
greenlet==2.0.2
griffe==0.25.5 ; python_version >= '3.7'
h11==0.14.0 ; python_version >= '3.7'
h2==4.1.0
hpack==4.0.0 ; python_full_version >= '3.6.1'
httpcore==0.16.3 ; python_version >= '3.7'
httpx[http2]==0.23.3 ; python_version >= '3.7'
hyperframe==6.0.1 ; python_full_version >= '3.6.1'
idna==3.4 ; python_version >= '3.5'
ipython==8.11.0
jedi==0.18.2 ; python_version >= '3.6'
jinja2==3.1.2 ; python_version >= '3.7'
jmespath==1.0.1 ; python_version >= '3.7'
jsonpatch==1.32 ; python_version >= '2.7' and python_version not in '3.0, 3.1, 3.2, 3.3, 3.4'
jsonpointer==2.3 ; python_version >= '2.7' and python_version not in '3.0, 3.1, 3.2, 3.3'
jsonschema==4.17.3 ; python_version >= '3.7'
kubernetes==26.1.0 ; python_version >= '3.6'
mako==1.2.4 ; python_version >= '3.7'
markdown==3.4.1 ; python_version >= '3.7'
markdown-it-py==2.2.0 ; python_version >= '3.7'
markupsafe==2.1.2 ; python_version >= '3.7'
matplotlib-inline==0.1.6 ; python_version >= '3.5'
mdurl==0.1.2 ; python_version >= '3.7'
multidict==6.0.4 ; python_version >= '3.7'
oauthlib==3.2.2 ; python_version >= '3.6'
orjson==3.8.7 ; python_version >= '3.7'
packaging==23.0 ; python_version >= '3.7'
parso==0.8.3 ; python_version >= '3.6'
pathspec==0.11.0 ; python_version >= '3.7'
pendulum==2.1.2 ; python_version >= '2.7' and python_version not in '3.0, 3.1, 3.2, 3.3, 3.4'
pexpect==4.8.0 ; sys_platform != 'win32'
pickleshare==0.7.5
prefect==2.8.3
prompt-toolkit==3.0.38 ; python_full_version >= '3.7.0'
ptyprocess==0.7.0
pure-eval==0.2.2
pyasn1==0.4.8
pyasn1-modules==0.2.8
pycparser==2.21
pydantic==1.10.5 ; python_version >= '3.7'
pygments==2.14.0 ; python_version >= '3.6'
pyrsistent==0.19.3 ; python_version >= '3.7'
python-dateutil==2.8.2 ; python_version >= '2.7' and python_version not in '3.0, 3.1, 3.2, 3.3'
python-slugify==8.0.1 ; python_version >= '3.7'
pytz==2022.7.1
pytz-deprecation-shim==0.1.0.post0 ; python_version >= '2.7' and python_version not in '3.0, 3.1, 3.2, 3.3, 3.4, 3.5'
pytzdata==2020.1 ; python_version >= '2.7' and python_version not in '3.0, 3.1, 3.2, 3.3'
pyyaml==6.0 ; python_version >= '3.6'
readchar==4.0.3 ; python_version >= '3.6'
regex==2022.10.31 ; python_version >= '3.6'
requests==2.28.2 ; python_version >= '3.7' and python_version < '4'
requests-oauthlib==1.3.1 ; python_version >= '2.7' and python_version not in '3.0, 3.1, 3.2, 3.3'
rfc3986[idna2008]==1.5.0
rich==13.3.2 ; python_full_version >= '3.7.0'
rsa==4.9 ; python_version >= '3.6'
s3fs==2023.3.0
setuptools==67.5.0 ; python_version >= '3.7'
six==1.16.0 ; python_version >= '2.7' and python_version not in '3.0, 3.1, 3.2, 3.3'
sniffio==1.3.0 ; python_version >= '3.7'
sqlalchemy[asyncio]==1.4.46 ; python_version >= '2.7' and python_version not in '3.0, 3.1, 3.2, 3.3, 3.4, 3.5'
stack-data==0.6.2
starlette==0.25.0 ; python_version >= '3.7'
text-unidecode==1.3
toml==0.10.2 ; python_version >= '2.6' and python_version not in '3.0, 3.1, 3.2, 3.3'
traitlets==5.9.0 ; python_version >= '3.7'
typer==0.7.0 ; python_version >= '3.6'
typing-extensions==4.5.0 ; python_version >= '3.7'
tzdata==2022.7 ; python_version >= '3.6'
tzlocal==4.2 ; python_version >= '3.6'
urllib3==1.26.14 ; python_version >= '2.7' and python_version not in '3.0, 3.1, 3.2, 3.3, 3.4, 3.5'
uvicorn==0.20.0 ; python_version >= '3.7'
wcwidth==0.2.6
websocket-client==1.5.1 ; python_version >= '3.7'
websockets==10.4 ; python_version >= '3.7'
wrapt==1.15.0 ; python_version >= '2.7' and python_version not in '3.0, 3.1, 3.2, 3.3, 3.4'
yarl==1.8.2 ; python_version >= '3.7'
```

**Comment by LeeMendelowitz** [comment]
If the deployment has a `path` argument provided, that now needs a trailing slash as well:

```
my_deployment = Deployment.build_from_flow(
    flow = my_flow,
    name = 'my-deployment',
    output = 'deployment.yaml',
    storage = storage,
    infrastructure = infrastructure,
    skip_upload = True,
    path = 'my-path/',
    apply = False
)
```

**Comment by Ishankoradia** [comment]
I am still facing this issue. I am on Prefect 2.8.3. I am generating deployments as follows 

```
flows/Flow.py
# Custom flow class that defines the flow
class Flow(BaseModel):
    
    @flow(name='custom_name')
    def my_func(self):
        print('in my flow function')

```
```
flow_function = getattr(flow, 'my_func') # Callable
# deployment
deployment = await Deployment.build_from_flow(
                    flow=flow_function.with_options(name='my_func_flow'),
                    name=deployment_name,
                    work_queue_name=work_queue_name,
                    tags = tags,
                )
deployment.apply()
```

The error that I face when I run the deployment manually from the UI 

```
Flow could not be retrieved from deployment.
Traceback (most recent call last):
  File "/Users/dorjayyolmo/Dev/data/DDP/ddp_prefect_starter/prefect-venv/lib/python3.10/site-packages/prefect/flows.py", line 809, in load_flow_from_entrypoint
    flow = import_object(entrypoint)
  File "/Users/dorjayyolmo/Dev/data/DDP/ddp_prefect_starter/prefect-venv/lib/python3.10/site-packages/prefect/utilities/importtools.py", line 210, in import_object
    return getattr(module, object_name)
AttributeError: module '__prefect_loader__' has no attribute 'my_func'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Users/dorjayyolmo/Dev/data/DDP/ddp_prefect_starter/prefect-venv/lib/python3.10/site-packages/prefect/engine.py", line 270, in retrieve_flow_then_begin_flow_run
    flow = await load_flow_from_flow_run(flow_run, client=client)
  File "/Users/dorjayyolmo/Dev/data/DDP/ddp_prefect_starter/prefect-venv/lib/python3.10/site-packages/prefect/client/utilities.py", line 47, in with_injected_client
    return await fn(*args, **kwargs)
  File "/Users/dorjayyolmo/Dev/data/DDP/ddp_prefect_starter/prefect-venv/lib/python3.10/site-packages/prefect/deployments.py", line 188, in load_flow_from_flow_run
    flow = await run_sync_in_worker_thread(load_flow_from_entrypoint, str(import_path))
  File "/Users/dorjayyolmo/Dev/data/DDP/ddp_prefect_starter/prefect-venv/lib/python3.10/site-packages/prefect/utilities/asyncutils.py", line 91, in run_sync_in_worker_thread
    return await anyio.to_thread.run_sync(
  File "/Users/dorjayyolmo/Dev/data/DDP/ddp_prefect_starter/prefect-venv/lib/python3.10/site-packages/anyio/to_thread.py", line 31, in run_sync
    return await get_asynclib().run_sync_in_worker_thread(
  File "/Users/dorjayyolmo/Dev/data/DDP/ddp_prefect_starter/prefect-venv/lib/python3.10/site-packages/anyio/_backends/_asyncio.py", line 937, in run_sync_in_worker_thread
    return await future
  File "/Users/dorjayyolmo/Dev/data/DDP/ddp_prefect_starter/prefect-venv/lib/python3.10/site-packages/anyio/_backends/_asyncio.py", line 867, in run
    result = context.run(func, *args)
  File "/Users/dorjayyolmo/Dev/data/DDP/ddp_prefect_starter/prefect-venv/lib/python3.10/site-packages/prefect/flows.py", line 811, in load_flow_from_entrypoint
    raise MissingFlowError(
prefect.exceptions.MissingFlowError: Flow function with name 'my_func_flow' not found in 'flows/Flow.py'.
```

**Comment by Samreay** [comment]
If you didn't have a path specified before, you will need to add a `path="/"` to get it working on this version of s3fs

**Comment by madkinsz** [comment]
Thanks for the report! It may be worth raising this upstream as it's not clear if this was an intended regression by them? I don't see anything in their changelog mentioning this.

**Comment by LeeMendelowitz** [comment]
I think the issue may be with a dependency of `s3fs` because I didn't see any noteworthy changes in s3fs between `2023.3.0` and `2023.1.0`. I wasn't exactly sure how to report the problem upstream since I'm not familiar with these libraries.

**Comment by madkinsz** [comment]
@LeeMendelowitz https://github.com/fsspec/filesystem_spec or https://github.com/fsspec/s3fs would be my recommendation.

**Comment by rdenham** [comment]
do you think this is related to [this issue](https://github.com/PrefectHQ/prefect-aws/issues/217)?

**Comment by madkinsz** [comment]
@rdenham I'm not sure.


I think this part of the diff looks suspicious: https://github.com/fsspec/filesystem_spec/compare/2023.1.0...2023.3.0#diff-9e28a72564624d8305a58b77c04c514d15e016b61899fd507b52b30388ef6ab6L343-R348

@LeeMendelowitz can you confirm that this error does not occur on fsspec `2023.1.0`?

**Comment by LeeMendelowitz** [comment]
@madkinsz I can confirm that the error does not occur with `fsspec==2023.1.0`, but the error does occur with `fsspec==2023.3.0`. 

I was able to combine `s3f3==2023.3.0` and `fsspec==2023.1.0` with:

```
pip install s3fs==2023.3.0
pip install --no-deps fsspec==2023.1.0
```

**Comment by martindurant** [comment]
Could you please at least specify what kind of operation was being done, and in what way it produced results differently from expected (or any exception within s3fs/fsspec). The "s3fs" logger is available which might give you more information.

> By simply adding a trailing slash to the S3.bucket_path, everything seems to work again.

There was a refactor in get/put/cp in fsspec trying for consistency between the backends and posix shells. Additional tests were added to s3fs, but none were broken - so your specific usage was probably not previously covered.

**Comment by madkinsz** [comment]
@martindurant definitely! Sorry it's not particularly clear what the problem is here, I'm not quite sure myself since there are a few steps involved. Here, their workflow is being "deployed" to an S3 bucket by uploading the working directory with repeated `filesystem.put_file` calls. Then, at runtime, we download the the directory using `filesystem.get`. Without a trailing `/`, the worfklow file is missing from the downloaded directory. I admit, we're doing some complicated path parsing of our own so we could definitely be doing something weird.

I'll see if I can find someone on our team to reproduce this so we can get more information.

Lee, is the working directory present in your bucket as expected? What does the working directory at `/Users/lmendelowitz/playground/prefect/storage_issue/working_dir/` look like when the flow retrieval fails?

**Comment by madkinsz** [comment]
fwiw I've tried to reproduce this with a local file system instead of an S3 bucket and could not do so.

**Comment by martindurant** [comment]
Local directories are distinct from files and always definitely exist or not. The trailing "/" to s3fs can be seen as a sign that you mean a "directory" path, not file - these isn't really another way to be specific.

**Comment by prefectcboyd** [comment]
I reproduced this with debug logging - 

A successful case (using a full path of `/tmp/flows/`) and the associated logs - 
```
11:08:52.862 | DEBUG   | prefect.infrastructure.process - Process 'gabby-toucan' running command: /Users/christopherboyd/virtualenvs/prefect2/bin/python -m prefect.engine in /var/folders/_s/1vczgxc97y5003d22b40kpd80000gn/T/tmpfvq0yvtcprefect
11:08:52.944 | INFO    | prefect.agent - Completed submission of flow run '5f289cf5-4909-46fe-ba34-2f4dab80f28b'
11:08:54.350 | INFO    | Flow run 'gabby-toucan' - Downloading flow code from storage at '/tmp/flows/'
11:08:54.474 | DEBUG   | s3fs - Setting up s3fs instance
11:08:54.795 | DEBUG   | s3fs - CALL: get_object - () - {'Bucket': 'chatbot', 'Key': 'tmp/flows/.prefectignore'}
11:08:54.810 | DEBUG   | s3fs - CALL: get_object - () - {'Bucket': 'chatbot', 'Key': 'tmp/flows/test_deploy.py'}
11:08:54.999 | DEBUG   | Flow run 'gabby-toucan' - Importing flow code from 'test_deploy.py:double_num'
11:08:55.942 | DEBUG   | s3fs - CALL: put_object - () - {'Bucket': 'chatbot', 'Key': 'tmp/flows/.prefectignore', 'overwrite': True}
11:08:56.005 | DEBUG   | s3fs - CALL: put_object - () - {'Bucket': 'chatbot', 'Key': 'tmp/flows/test_deploy.py', 'overwrite': True, 'ContentType': 'text/x-python'}
11:08:56.106 | DEBUG   | s3fs - CALL: put_object - () - {'Bucket': 'chatbot', 'Key': 'tmp/flows/__pycache__/test_deploy.cpython-39.pyc', 'overwrite': True, 'ContentType': 'application/x-python-code'}
11:08:56.824 | DEBUG   | Flow run 'gabby-toucan' - Starting 'ConcurrentTaskRunner'; submitted tasks will be run concurrently...
11:08:57.515 | INFO    | Flow run 'gabby-toucan' -
11:08:57.673 | INFO    | Flow run 'gabby-toucan' - Finished in state Completed()
11:08:58.100 | INFO    | prefect.infrastructure.process - Process 'gabby-toucan' exited cleanly.
```

Simply removing the trailing slash (so `/tmp/flows`) yields the following:
```
16:20:23.401 | INFO    | prefect.agent - Submitting flow run 'c08f2854-414d-41f7-b744-310bd3c180c6'
16:20:23.868 | INFO    | prefect.infrastructure.process - Opening process 'massive-locust'...
16:20:23.875 | DEBUG   | prefect.infrastructure.process - Process 'massive-locust' running command: /Users/christopherboyd/virtualenvs/prefect2/bin/python -m prefect.engine in /var/folders/_s/1vczgxc97y5003d22b40kpd80000gn/T/tmp36y_o03_prefect
16:20:23.964 | INFO    | prefect.agent - Completed submission of flow run 'c08f2854-414d-41f7-b744-310bd3c180c6'
16:20:25.184 | INFO    | Flow run 'massive-locust' - Downloading flow code from storage at '/tmp/flows'
16:20:25.322 | DEBUG   | s3fs - Setting up s3fs instance
16:20:25.650 | DEBUG   | s3fs - CALL: get_object - () - {'Bucket': 'chatbot', 'Key': 'tmp/flows/.prefectignore'}
16:20:25.653 | DEBUG   | s3fs - CALL: get_object - () - {'Bucket': 'chatbot', 'Key': 'tmp/flows/test_deploy.py'}
16:20:25.888 | DEBUG   | Flow run 'massive-locust' - Importing flow code from 'test_deploy.py:double_num'
16:20:25.893 | ERROR   | Flow run 'massive-locust' - Flow could not be retrieved from deployment.
Traceback (most recent call last):
  File "<frozen importlib._bootstrap_external>", line 846, in exec_module
  File "<frozen importlib._bootstrap_external>", line 982, in get_code
  File "<frozen importlib._bootstrap_external>", line 1039, in get_data
FileNotFoundError: [Errno 2] No such file or directory: 'test_deploy.py'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Users/christopherboyd/virtualenvs/prefect2/lib/python3.9/site-packages/prefect/engine.py", line 268, in retrieve_flow_then_begin_flow_run
    flow = await load_flow_from_flow_run(flow_run, client=client)
  File "/Users/christopherboyd/virtualenvs/prefect2/lib/python3.9/site-packages/prefect/client/utilities.py", line 47, in with_injected_client
    return await fn(*args, **kwargs)
  File "/Users/christopherboyd/virtualenvs/prefect2/lib/python3.9/site-packages/prefect/deployments.py", line 187, in load_flow_from_flow_run
    flow = await run_sync_in_worker_thread(load_flow_from_entrypoint, str(import_path))
  File "/Users/christopherboyd/virtualenvs/prefect2/lib/python3.9/site-packages/prefect/utilities/asyncutils.py", line 91, in run_sync_in_worker_thread
    return await anyio.to_thread.run_sync(
  File "/Users/christopherboyd/virtualenvs/prefect2/lib/python3.9/site-packages/anyio/to_thread.py", line 31, in run_sync
    return await get_asynclib().run_sync_in_worker_thread(
  File "/Users/christopherboyd/virtualenvs/prefect2/lib/python3.9/site-packages/anyio/_backends/_asyncio.py", line 937, in run_sync_in_worker_thread
    return await future
  File "/Users/christopherboyd/virtualenvs/prefect2/lib/python3.9/site-packages/anyio/_backends/_asyncio.py", line 867, in run
    result = context.run(func, *args)
  File "/Users/christopherboyd/virtualenvs/prefect2/lib/python3.9/site-packages/prefect/flows.py", line 786, in load_flow_from_entrypoint
    flow = import_object(entrypoint)
  File "/Users/christopherboyd/virtualenvs/prefect2/lib/python3.9/site-packages/prefect/utilities/importtools.py", line 197, in import_object
    module = load_script_as_module(script_path)
  File "/Users/christopherboyd/virtualenvs/prefect2/lib/python3.9/site-packages/prefect/utilities/importtools.py", line 160, in load_script_as_module
    raise ScriptError(user_exc=exc, path=path) from exc
prefect.exceptions.ScriptError: Script at 'test_deploy.py' encountered an exception: FileNotFoundError(2, 'No such file or directory')
16:20:26.386 | INFO    | prefect.infrastructure.process - Process 'massive-locust' exited cleanly.
```

**Comment by martindurant** [comment]
It seems to be trying the GET the exact same file key in each case, so I don't know why the second case would fail.

**Comment by Hongbo-Miao** [comment]
Just add extra info, this issue blocks up upgrading Prefect after Prefect 2.8.4. It would be great to let Prefect works well with latest fsspec 2023.3.0, thanks! 😃

Because we get error

> Because prefect (2.8.5) depends on fsspec (>=2022.5.0,<2023.3.0 || >2023.3.0)
 and s3fs (2023.3.0) depends on fsspec (2023.3.0), prefect (2.8.5) is incompatible with s3fs (2023.3.0).
So, because restore-influxdb-v2 depends on both prefect (2.8.5) and s3fs (2023.3.0), version solving failed.

**Comment by LeeMendelowitz** [comment]
See https://github.com/fsspec/s3fs/issues/717

**Comment by madkinsz** [comment]
@Hongbo-Miao we agree! We're waiting for the cause of this to be determined as it should probably be addressed upstream.

@LeeMendelowitz thank you so much!

**Comment by madkinsz** [comment]
Closed by #9440
